platform/distance.cpp

```diff
@@ -0,0 +1,246 @@
+#include "distance.hpp"
+
+#include "platform/localization.hpp"
+#include "platform/measurement_utils.hpp"
+
+#include "base/assert.hpp"
+
+namespace platform
+{
+namespace
+{
+Distance MetersTo(double distance, Distance::Units units)
+{
+  switch (units)
+  {
+  case Distance::Units::Kilometers: return Distance(distance / 1000, Distance::Units::Kilometers);
+  case Distance::Units::Feet:
+    return Distance(measurement_utils::MetersToFeet(distance), Distance::Units::Feet);
+  case Distance::Units::Miles:
+    return Distance(measurement_utils::MetersToMiles(distance), Distance::Units::Miles);
+  default: UNREACHABLE();
+  }
+}
+
+Distance KilometersTo(double distance, Distance::Units units)
+{
+  return MetersTo(distance * 1000, units);
+}
+
+Distance FeetTo(double distance, Distance::Units units)
+{
+  switch (units)
+  {
+  case Distance::Units::Meters:
+    return Distance(measurement_utils::FeetToMeters(distance), Distance::Units::Meters);
+  case Distance::Units::Kilometers:
+    return Distance(measurement_utils::FeetToMeters(distance) / 1000, Distance::Units::Kilometers);
+  case Distance::Units::Miles:
+    return Distance(measurement_utils::FeetToMiles(distance), Distance::Units::Miles);
+  default: UNREACHABLE();
+  }
+}
+
+Distance MilesTo(double distance, Distance::Units units)
+{
+  switch (units)
+  {
+  case Distance::Units::Meters:
+    return Distance(measurement_utils::MilesToMeters(distance), Distance::Units::Meters);
+  case Distance::Units::Kilometers:
+    return Distance(measurement_utils::MilesToMeters(distance) / 1000, Distance::Units::Kilometers);
+  case Distance::Units::Feet:
+    return Distance(measurement_utils::MilesToFeet(distance), Distance::Units::Feet);
+  default: UNREACHABLE();
+  }
+}
+
+Distance NauticalMilesTo(double distance, Distance::Units units)
+{
+  switch (units)
+  {
+  case Distance::Units::Meters:
+    return Distance(measurement_utils::NauticalMilesToMeters(distance), Distance::Units::Meters);
+  default: UNREACHABLE();
+  }
+}
+
+Distance InchesTo(double distance, Distance::Units units)
+{
+  switch (units)
+  {
+  case Distance::Units::Meters:
+    return Distance(measurement_utils::InchesToMeters(distance), Distance::Units::Meters);
+  default: UNREACHABLE();
+  }
+}
+
+double WithPrecision(double value, uint8_t precision)
+{
+  std::ostringstream ss;
+  ss << std::setprecision(precision) << std::fixed << value;
+  return std::atof(ss.str().c_str());
```

Comment:
stod or better fast double converter from base library (we'll later replace it with even more faster converter).

Reply:
Good here if precision matters.